src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,117 @@
-import React from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { postComment } from '../api/ClientFunctions';
+import { Comment } from '../types/Comment';
+
+interface Props {
+  postId: number;
+  commentsError: boolean;
+  setCommentsError: (value: boolean) => void;
+  setDoesFormExist: (value: boolean) => void;
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  commentsError,
+  setCommentsError,
+  setDoesFormExist,
+  setComments,
+}) => {
+  // #region states
+
+  const [isNameValid, setIsNameValid] = useState(true);
+  const [isEmailValid, setIsEmailValid] = useState(true);
+  const [isCommentValid, setIsCommentValid] = useState(true);
+  const [nameValue, setNameValue] = useState('');
+  const [emailValue, setEmailValue] = useState('');
+  const [commentValue, setCommentValue] = useState('');
+  const [isSubmitLoading, setIsSubmitLoading] = useState(false);
+
+  // #endregion
+  // #region handlers
+
+  const clearHandler = () => {
+    setNameValue('');
+    setEmailValue('');
+    setCommentValue('');
+    setIsNameValid(true);
+    setIsEmailValid(true);
+    setIsCommentValid(true);
+  };
+
+  const onSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    const IS_VALID =
+      nameValue.length > 0 && emailValue.length > 0 && commentValue.length > 0;
+
+    if (nameValue.length === 0) {
+      setIsNameValid(false);
+    }
+
+    if (emailValue.length === 0) {
+      setIsEmailValid(false);
+    }
+
+    if (commentValue.length === 0) {
+      setIsCommentValid(false);
+    }
+
+    if (!IS_VALID) {
+      return;
+    }
+
+    setIsSubmitLoading(true);
+
+    const comment = {
+      postId: postId,
+      name: nameValue,
+      email: emailValue,
+      body: commentValue,
+    };
+
+    if (commentsError) {
+      setCommentsError(false);
+    }
+
+    postComment(comment)
+      .then(_comment => {
+        setComments(currentComments => [...currentComments, _comment]);
+        setCommentValue('');
+      })
+      .catch(() => {
+        setCommentsError(true);
+        setDoesFormExist(false);
```

Comment:
When an error occurs during the comment submission, `setDoesFormExist(false)` is called, which might hide the form. Consider keeping the form visible to allow users to correct their input and try again.